boot/build.go

```diff
@@ -113,20 +113,28 @@ func (b Build) Build(context libcnb.BuildContext) (libcnb.BuildResult, error) {
 	}
 
 	buildNativeImage := false
+	nativeImageArgFile := ""
 	if n, ok, err := pr.Resolve("spring-boot"); err != nil {
 		return libcnb.BuildResult{}, fmt.Errorf("unable to resolve spring-boot plan entry\n%w", err)
 	} else if ok {
 		if v, ok := n.Metadata["native-image"].(bool); ok {
 			buildNativeImage = v
 		}
 	}
+	if _, ok, err := pr.Resolve("spring-boot-native"); err != nil {
```

Comment:
What do you think about a slightly different name for the plan entry?

- `spring-boot-native-args-file`
- `native-image-args-file`

I haven't thought this through all the way but this code is a little confusing because we're looking for basically the same thing in the build plan two different ways, once with metadata once with a specific buildplan. It seems like the new plan entry is used for locating the args file, so maybe the plan name can be more specific to that. If it can be generic, even better.

Reply:
> we're looking for basically the same thing in the build plan two different ways

That's exactly what we're doing, unfortunately because the plan requirement in `native-image` is tied to the environment variable.
Agreed on the generic plan name, since it must be referenced in `native-image` and another buildpack could use this to provide an argfile and trigger a native build. I'll go with `native-image-args-file`?